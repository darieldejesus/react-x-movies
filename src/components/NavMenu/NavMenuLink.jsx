import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'styled-components';
import { NavLink } from 'react-router-dom';

const Link = Styled(NavLink)`
  color: #63676f;
  font-size: 15px;
  font-weight: 500;
  text-decoration: none;
  text-transform: uppercase;
`;

Link.displayName = 'Link';

const NavItem = Styled.li`
  background: #eaeaeb;
  box-shadow: inset -1px 0px 0px #cecece;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  flex: 1;
  justify-content: center;
  list-style-type: none;
  padding: 10px 20px;
  text-align: center;

  &:last-child {
    box-shadow: none;	
  }`;

NavItem.displayName = 'NavItem';

const NavMenuLink = ({ href, label }) => (
  <NavItem>
    <Link to={href} activeClassName="active">
      {label}
    </Link>
  </NavItem>
);

NavMenuLink.propTypes = {
  href: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default NavMenuLink;