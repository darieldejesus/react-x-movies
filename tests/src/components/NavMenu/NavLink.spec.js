import React from 'react';
import NavMenuLink from '../../../../src/components/NavMenu/NavMenuLink';

describe('NavMenuLink', function () {
  beforeEach(() => {
    this.wrapper = shallow(<NavMenuLink href="test_href" label="test_label" />);
  });
  test('should renders NavLinks', () => {
    expect(toJson(this.wrapper)).toMatchSnapshot();
  });
});
