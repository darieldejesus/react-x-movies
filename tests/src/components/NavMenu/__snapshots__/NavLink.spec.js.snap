// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenuLink should renders NavLinks 1`] = `
<NavItem>
  <Link
    activeClassName="active"
    to="test_href"
  >
    test_label
  </Link>
</NavItem>
`;
